src/blocked.c

```diff
@@ -624,7 +636,23 @@ static void unblockClientOnKey(client *c, robj *key) {
      * we need to re process the command again */
     if (c->flags & CLIENT_PENDING_COMMAND) {
         c->flags &= ~CLIENT_PENDING_COMMAND;
+        /* We must set the current client here so it will be available
+         * when we will try to send the the client side caching notification
+         * of 'afterCommand'. */
+        client *old_client = server.current_client;
```

Comment:
any reason we explicitly do all the current_client+ after_command and nesting arithmetic? AFAIK this is already done in processCommandAndResetClient

Reply:
@ranshid I believe my comment bellow answer that. We want the reprocessing of the command and the command handler to be atomic.

Reply:
I will try to clarify that in the comment, thanks.

Reply:
Let me know if its better now.

Reply:
Yes. Thank you!